src/NUnitFramework/framework/Internal/AwaitAdapter.cs

```diff
@@ -0,0 +1,111 @@
+// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+
+#if ASYNC
```

Comment:
The `AwaitAdapter` is only used in `AsyncInvocationRegion` which is only used when `ASYNC` is defined, so why isn't the whole class wrapped in `#if ASYNC` and why the `#ifdef` for other platforms that don't support async? Future PR?

Reply:
By other platform `#ifdef`s, I mean in the following files which could also be wrapped in an `#if ASYNC`.

Reply:
🗨 https://github.com/nunit/nunit/pull/2776#discussion_r177935936

Reply:
A future PR might remove the need for most of our #if ASYNC, but that should be done atomically.